### ISSUE TITLE
pymatgen==2020.9.14: PDPlotter cannot write image (AttributeError)

### ISSUE THREAD

**Issue by LuciusV** [materialsproject/pymatgen#1966]
**Describe the bug**
If I try to .write_image() of ternary phase diagram with default backend, it fails.

**To Reproduce**
In jupyter notebook:
1. Make some 3-component PhaseDiagram 
2. Make PDPlotter on this Phasediagram
3. Call write_image('something.svg'). PNG also fails.
4. it fails (AttributeError: 'Figure' object has no attribute 'gcf')  apparently expecting matplotlib object...

**Expected behavior**
SVG Image saved normally

**Desktop (please complete the following information):**
 - OS: Linux 
 - Version: 2020.9.14

**Additional context**

I know that I can save png from resulting plotly plot, but I need svg :(

**Comment by mattmcdermott** [comment]
@LuciusV My apologies for just seeing this now! Yes, the new PDPlotter (with Plotly backend) does not currently work with the `write_image()` function; that's my fault for missing that implementation when I wrote the code!

Unfortunately, Plotly is quite difficult when it comes to saving high-quality figures, but I've identified a workaround for now. (using some PhaseDiagram object named `pd`):

```
fig = PDPlotter(pd).get_plot()
config = {
  'toImageButtonOptions': {
    'format': 'svg', 
  }
}

fig.write_html("my_img.html", config=config)
```
This changes the image save button to be SVG, then creates an HTML file that you can open in your browser and adjust to your liking before clicking save. I find this to be the most convenient method as of now.

I'll look into fixing the `write_image()` function next week... I know I've had some issues with some of the Plotly saving methods in the past.

**Comment by mkhorton** [comment]
Yes, to write images you need to have `orca` installed (typically) and use `plotly.io`, writing to html is another way to do it. The former solution is probably the better one, though that will require the extra dependency for people who want to write images out with the plotly backend.

**Comment by LuciusV** [comment]
This is very good workaround for my case, thank you very much!

**Comment by TholeoR** [comment]
For me the error still exists, but I found a way that works for me.
I modified the PDPlotter.write_image function to:

```
def write_image(self, stream, image_format="svg", **kwargs):
        
        plot = self.get_plot(**kwargs)
        plot.write_image(stream, format=image_format, width=1073, height=700, scale=1)
 ```
It uses the function plotly.io.write_image() instead of plt.savefig().

**Comment by mattmcdermott** [comment]
@TholeoR This has been addressed with #3032. Apologies for the delay!
